src/main/java/bio/terra/service/profile/flight/create/ProfileCreateFlight.java

```diff
@@ -24,13 +25,16 @@ public ProfileCreateFlight(FlightMap inputParameters, Object applicationContext)
 
     AuthenticatedUserRequest user =
         inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    Optional<String> idpAccessToken =
+        Optional.ofNullable(
+            inputParameters.get(JobMapKeys.IDP_ACCESS_TOKEN.getKeyName(), String.class));
 
     CloudPlatformWrapper platform = CloudPlatformWrapper.of(request.getCloudPlatform());
 
     addStep(new GetOrCreateProfileIdStep(request));
     addStep(new CreateProfileMetadataStep(profileService, request, user));
     if (platform.isGcp()) {
-      addStep(new CreateProfileVerifyAccountStep(profileService, request, user));
+      addStep(new CreateProfileVerifyAccountStep(profileService, request, user, idpAccessToken));
```

Comment:
FWIW, we already technically have this info in the User object (which includes the JWT token).  Would it make sense to just have a method that extracts the google token from a user object?

Reply:
I want to avoid parsing the JWT. Doing it correctly requires checking things like it is signed by a sig we trust, has the correct issuer and audience and maybe other things. Appsec cares that we do it right so I would prefer to side step that scrutiny and let the proxy do it because it is doing it anyway.

Reply:
Won't the proxy block the request if it's an invalid JWT with that very mechanism?  We can assume that when it reaches us it's a valid JWT

Reply:
I will also mention, AuthenticatedUserRequests are deprecated in favor of direct token access and I wouldn't want to introduce anything that ties us more closely to them.

Reply:
Right, I think it still holds.  The flights will need two objects instead of just `AuthenticatedUserRequests` once we move in that direction and we'd be able to get the idp from the JWT directly still

Reply:
I agree that we should be able to assume the JWT is valid. But I think there are enough JWT attack vectors that in the past, my read from appsec was always do it right.

Reply:
I have posed the question again to get a definitive answer

Reply:
Got it.  I was just thinking passing more information around than we have to would be problematic too but if appsec is bought in then go for it

Reply:
I stand corrected. I am going to undo most of this code and parse the JWT in `GoogleBillingService.accessClient(AuthenticatedUserRequest user)`